src/ConfigPostProcessor.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * @see       https://github.com/laminas/laminas-zendframework-bridge for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-zendframework-bridge/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-zendframework-bridge/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\ZendFrameworkBridge;
+
+class ConfigPostProcessor
+{
+    /** @var Replacements */
+    private $replacements;
+
+    public function __construct()
+    {
+        $this->replacements = new Replacements([
+            // Since we're rewriting at the PHP level, and not the raw
+            // strings from reading file contents, we need to add this
+            // one.
+            'zend-expressive' => 'expressive',
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    public function __invoke(array $config)
+    {
+        $rewritten = [];
+
+        foreach ($config as $key => $value) {
+            $newKey = is_string($key) ? $this->replacements->replace($key) : $key;
+
+            if (isset($rewritten[$newKey]) && is_array($rewritten[$newKey])) {
+                $rewritten[$newKey] = self::merge($rewritten[$newKey], $this->rewriteValue($value, $newKey));
+                continue;
+            }
+
+            $rewritten[$newKey] = $this->rewriteValue($value, $newKey);
+        }
+
+        return $rewritten;
+    }
+
+    /**
+     * Perform subsitutions as needed on an individual value.
+     *
+     * The $key is provided to ensure we do not rewrite aliases configuration.
+     *
+     * @param mixed $value
+     * @param string|int $key
+     * @return mixed
+     */
+    private function rewriteValue($value, $key)
+    {
+        if (is_string($value)) {
+            return $this->replacements->replace($value);
+        }
+
+        if (null === $value || is_scalar($value) || is_object($value)) {
+            return $value;
+        }
+
+        // Array; time to recurse, but only if not aliases.
+        return $key === 'aliases' ? $value : $this($value);
```

Comment:
we don't want to rewrite sub-keys of `aliases` arrays, but I think we should rewrite values, for example:

```php
'aliases' => [
    'Zend\Package\MyAlias' => 'Zend\Package\AwesomeClass',
],
```

we want to become:
```php
'aliases' => [
    'Zend\Package\MyAlias' => 'Laminas\Package\AwesomeClass',
],
```

it will reduce iterations in SM - without rewriting it should also work, unless... this very edge case but still possible:
```php
'factories' => [
    'Zend\Something' => Zend\...\InvokableFactory::class,
],
'aliases' => [
   'my-alias' => '\Zend\Something',
],
```

in current state we are gonna get:

```php
'factories' => [
    'Laminas\Something' => Laminas\...\InvokableFactory::class,
],
'aliases' => [
   'my-alias' => '\Zend\Something',
],
```

and calling `$sm->get('my-alias')` is gonna fail, as we do not have defined `Zend\Something` anymore.

Reply:
That's a great test case for me - I'll get to work on this shortly.